python/GafferSceneTest/CustomAttributesTest.py

```diff
@@ -291,5 +291,34 @@ def testAffects( self ) :
 
 		self.assertEqual( set( s["a"].affects( p["value"] ) ), set( [ s["a"]["out"]["attributes"], s["a"]["out"]["globals"] ] ) )
 
+	def testExtraAttributes( self ) :
+
+		s = Gaffer.ScriptNode()
+
+		s["sphere"] = GafferScene.Sphere()
+		s["a"] = GafferScene.CustomAttributes()
+		s["f"] = GafferScene.PathFilter()
+		s["f"]["paths"].setValue( IECore.StringVectorData( [ "/sphere" ] ) )
+		s["a"]["filter"].setInput( s["f"]["out"] )
+		s["a"]["extraAttributes"].setValue(IECore.CompoundData({
+			"a1" : IECore.StringData( "from extra" ),
+			"a2" : IECore.IntData( 2 ),
+		}))
+		s["a"]["attributes"].addChild(
+			Gaffer.NameValuePlug( "a1", IECore.StringData( "from attributes" ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic )
```

Comment:
This test needs tweaking for the backport, becacuse `NameValuePlug` doesn't exist yet.

```
======================================================================
ERROR: testExtraAttributes (GafferSceneTest.CustomAttributesTest.CustomAttributesTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/GafferHQ/gaffer/install/gaffer-0.53.6.2-linux/python/GafferSceneTest/CustomAttributesTest.py", line 310, in testExtraAttributes
    Gaffer.NameValuePlug( "a1", IECore.StringData( "from attributes" ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic )
AttributeError: 'module' object has no attribute 'NameValuePlug'
----------------------------------------------------------------------
```

Reply:
Fixed. Do we usually keep fixes like this in a separate commit to preserve the original that'll be found on master or do we eliminate all traces of this ever having been there?

I've also mentioned Tom's config in the Changes.

Reply:
I think the separate commit is fine, but I'm not sure we have a hard rule about it. Often times its difficult to separate because of merge conflicts.